arnold/crud/read/sample.py

```diff
@@ -18,4 +18,4 @@ def find_sample(adapter: ArnoldAdapter, sample_id: str) -> Optional[Sample]:
 def find_all_samples(adapter: ArnoldAdapter) -> List[Sample]:
     """Find all samples from the step collection"""
     raw_samples = adapter.sample_collection.find()
-    return parse_obj_as(List[Sample], list(raw_samples))
+    return [Sample(**raw_sample) for raw_sample in raw_samples]
```

Comment:
I think `Sample.model_validate(raw_sample)` might be recommended in Pydantic v2 but I might be wrong. Feel free to keep as is if you prefer this, though.